core/src/executor/alter/table.rs

```diff
@@ -23,7 +23,7 @@ pub async fn create_table<T: GStore + GStoreMut>(
         Some(Query { body, .. }) => match body {
             SetExpr::Select(select_query) => match &select_query.from.relation {
                 TableFactor::Table { name, .. } => {
-                    let schema = storage.fetch_schema(name).await?;
+                    let schema = storage.ok_or(AlterError::AlterUnsupported)?.fetch_schema(name).await?;
```

Comment:
`storage.ok_or(AlterError::AlterUnsupported)?` codes takes ownership of storage. Then it means that we cannot pass that to `select` which requires `Option<&T>`.

We can unwrap Option in the beginning.
```rust
let storage = storage.ok_or(AlterError::AlterUnsupported)?;
```

Then for `select`
```rust
let rows = select(Some(storage), query, None)
 ....
```

will work.